### PR TITLE
qos: T1871: add MTU option when configure limiter traffic-policy

### PR DESCRIPTION
**Pull request by HollyGurza**
## Change Summary

Added MTU option for qos limiter traffic policy
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T1871

## How to test

```
set qos policy limiter T1871 default bandwidth '10mbit'
set qos policy limiter T1871 default burst '100kb'
set qos policy limiter T1871 default mtu 10000
set qos interface eth0 ingress T1871 
```

## Smoketest result

```
vyos@vyos:~$ python3 /usr/libexec/vyos/tests/smoke/cli/test_qos.py 
test_01_cake (__main__.TestQoS.test_01_cake) ... ok
test_02_drop_tail (__main__.TestQoS.test_02_drop_tail) ... ok
test_03_fair_queue (__main__.TestQoS.test_03_fair_queue) ... ok
test_04_fq_codel (__main__.TestQoS.test_04_fq_codel) ... ok
test_05_limiter (__main__.TestQoS.test_05_limiter) ... ok
test_06_network_emulator (__main__.TestQoS.test_06_network_emulator) ... ok
test_07_priority_queue (__main__.TestQoS.test_07_priority_queue) ... ok
test_08_random_detect (__main__.TestQoS.test_08_random_detect) ... skipped 'tc returns invalid JSON here - needs iproute2 fix'
test_09_rate_control (__main__.TestQoS.test_09_rate_control) ... ok
test_10_round_robin (__main__.TestQoS.test_10_round_robin) ... ok
test_11_shaper (__main__.TestQoS.test_11_shaper) ... ok

----------------------------------------------------------------------
Ran 11 tests in 40.376s

OK (skipped=1)

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/include/qos/mtu.xml.i [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+<leafNode name="mtu">
+  <properties>
+    <help>MTU size for this class</help>
+    <valueHelp>
+      <format>&lt;number&gt;</format>
+      <description>Bytes</description>
+    </valueHelp>
+    <valueHelp>
+      <format>&lt;number&gt;&lt;suffix&gt;</format>
+      <description>Bytes with scaling suffix (kb, mb, gb)</description>
```
Comment (dmbaturin) [inline interface-definitions/include/qos/mtu.xml.i]: Does it actually accept gigabyte-large MTUs? Even if yes — should we allow that?

Comment (c-po) [reply interface-definitions/include/qos/mtu.xml.i]: > Does it actually accept gigabyte-large MTUs? Even if yes — should we allow that?

I'm unaware of a Gigabyte MTU 

Please use the common MTU definition instead: https://github.com/vyos/vyos-1x/blob/current/interface-definitions/include/interface/mtu-68-16000.xml.i

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/include/qos/mtu.xml.i (hunk 1 above)
Comment (HollyGurza) [reply interface-definitions/include/qos/mtu.xml.i]: 68 is too low for tc possible value is 256 - 4294967295
```
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 68
RTNETLINK answers: Cannot allocate memory
We have an error talking to the kernel
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 255
RTNETLINK answers: Cannot allocate memory
We have an error talking to the kernel
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 256
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 4294967295
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 4294967296
Error: argument "4294967296" is wrong: mtu
```

in the task description, author said `the packet size between ip stack and nic could be 65536` therefore range 68-16000 probably is incorrect for this case, I think we can use range 256-65535.

Comment (c-po) [reply interface-definitions/include/qos/mtu.xml.i]: Then please limit to this range

Comment (HollyGurza) [reply interface-definitions/include/qos/mtu.xml.i]: done

**Review by c-po — APPROVED** [review]

**Review by c-po — CHANGES_REQUESTED** [review]
Please also add the missing output of the smoketests and extend them with this new MTU option so it's always verified on image build.

https://github.com/vyos/vyos-1x/blob/current/smoketest/scripts/cli/test_qos.py

**Comment by HollyGurza** [comment]
> Please also add the missing output of the smoketests and extend them with this new MTU option so it's always verified on image build.
> 
> https://github.com/vyos/vyos-1x/blob/current/smoketest/scripts/cli/test_qos.py

done, added mtu and other missing options for limiter test.
also added mtu to `class` section, in the previous iteration it was only in the default section

Screenshot 2024-03-15 at 7 27 44 PM

File: smoketest/scripts/cli/test_qos.py [hunk 2]
```diff
@@ -279,6 +291,15 @@ def test_05_limiter(self):
...
                     self.cli_set(qos_class_base + ['bandwidth', qos_class_config['bandwidth']])
 
+                if 'exceed' in qos_class_config:
+                    self.cli_set(qos_class_base + ['exceed', qos_class_config['exceed']])
+
+                if 'burst' in qos_class_config:
+                    self.cli_set(qos_class_base + ['burst', qos_class_config['burst']])
+
+                if 'mtu' in qos_class_config:
+                    self.cli_set(qos_class_base + ['mtu', qos_class_config['mtu']])
```
Comment (c-po) [inline smoketest/scripts/cli/test_qos.py:308]: I see the MTU programmed to the kernel, but there is no validation who checks if the programmed value is the value when it's read back.

**Review by c-po — CHANGES_REQUESTED** [review]

File: smoketest/scripts/cli/test_qos.py (hunk 2 above)
Comment (HollyGurza) [reply smoketest/scripts/cli/test_qos.py:308]: done. Added validate what exactly is set to tc policy, not ideal but works.

**Review by c-po — APPROVED** [review]

**Review by sever-sever — APPROVED** [review]

**Comment by HollyGurza** [comment]
@c-po Need backport to Sagitta?

Screenshot 2024-03-20 at 1 05 19 PM

**Comment by c-po** [comment]
@MergifyIo backport sagitta

**Pull request merged**